### ISSUE TITLE
Provide documentation for _metadata feature

### ISSUE THREAD

**Issue by thjaeckle** [eclipse/ditto#785]
Was not done as part of #680, so must still be done.

**Comment by JulianFeinauer** [comment]
@thjaeckle and @jufickel-b I started to work on the documentation in this branch: https://github.com/JulianFeinauer/ditto/tree/feature/785-documentation-metadata do you agree with the current setup?

**Comment by thjaeckle** [comment]
@JulianFeinauer cool - I just had a brief look.
* adding it as a new page is good - as it is a new concept
* then you also must add it to the `_data/sidebards/ditto_sidebar.yml` or else it won't show in the navigation
* the current implementation does not have any predefined metadata, even not `issuedAt` - so that one also is not implicitly set
** I created #778 in order to bring `modified` and `revision` as predefined (and Ditto managed) metadata

I'm not sure if that will work:
```json
{
  "key":"*/",
  "value":{"issuedAt": someTimestamp }
}
```

I would have expected to look like:
```json
{
  "key": "*/issuedAt",
  "value": someTimestamp
}
```

**Comment by JulianFeinauer** [comment]
Hey @thjaeckle thanks for the feedback.. I did not finish it the weekent because I wanted to sit down and try it out myself as I did not track all changes you and @jufickel-b did in the code so before submitting it as PR I will try to test it.

**Comment by jufickel-b** [comment]
@thjaeckle your expectation is correct.
@JulianFeinauer you can have a look at the unit tests to see the format of metadata headers: [DefaultMetadataHeaderTest](https://github.com/eclipse/ditto/blob/3d326f065c08e185a7f945a9b9e5df125cd61c1d/model/base/src/test/java/org/eclipse/ditto/model/base/headers/metadata/DefaultMetadataHeaderTest.java)

**Comment by w4tsn** [comment]
This not only needs to be added to the doc pages but also to the swagger docs.

**Comment by w4tsn** [comment]
The API example / specification under [basic-thing.md](https://github.com/JulianFeinauer/ditto/blob/feature/785-documentation-metadata/documentation/src/main/resources/pages/ditto/basic-thing.md) needs to be updated to.

In the current docs: https://www.eclipse.org/ditto/basic-thing.html

**Comment by thjaeckle** [comment]
Closed via merged #806
